distributed/shuffle/shuffle.py

```diff
@@ -53,6 +54,8 @@ def rearrange_by_column_p2p(
     column: str,
     npartitions: int | None = None,
 ):
+    from dask.dataframe import DataFrame
```

Comment:
I don't love how `dask.dataframe.DataFrame` is also imported `if TYPE_CHECKING`. It means I could remove this line here, and the code would still pass mypy and flake8, even though it would fail at runtime. Tests would still fail, but I could see making a similar mistake in the future in some place that didn't have test coverage, and unwittingly introducing a runtime error.

However, because I want to use `DataFrame` as a type annotation, it does need to be imported for type-checkers.

I don't really think there's a way around this. Just curious if anyone had ideas, particularly @crusaderky or @ian-r-rose

Reply:
Yep, this issue is well-known upstream: https://github.com/PyCQA/pyflakes/pull/530